app/delegate-server/Types/AppExitReason.purs

```diff
@@ -1,15 +1,19 @@
 module DelegateServer.Types.AppExitReason
   ( AppExitReason
       ( AppExitReason_HeadFinalized
+      , AppExitReason_BiddingTimeExpired_UnexpectedHeadStatus
       )
   ) where
 
 import Prelude
 
 import Data.Generic.Rep (class Generic)
 import Data.Show.Generic (genericShow)
+import DelegateServer.Types.HydraHeadStatus (HydraHeadStatus)
 
-data AppExitReason = AppExitReason_HeadFinalized
+data AppExitReason
+  = AppExitReason_HeadFinalized
+  | AppExitReason_BiddingTimeExpired_UnexpectedHeadStatus HydraHeadStatus
```

Comment:
Is it one or two different reasons?

Reply:
It's two different reasons:

1. `AppExitReason_HeadFinalized`: Standing bid has been moved back to L1, final state has been reached, nothing left to be done.
2. `AppExitReason_BiddingTimeExpired_UnexpectedHeadStatus`: Bidding time expired, but the head has an unexpected status, that does not imply any further actions. It's actually only possible for `Idle` status, so I will update this constructor to `AppExitReason_BiddingTimeExpired_HeadIdle`.

Reply:
Updated `AppExitReason` in https://github.com/mlabs-haskell/hydra-auction-offchain/pull/35/commits/67a4a268b775a6613b177046aec4a49bc707c847